src/java.desktop/share/classes/sun/awt/image/ImageConsumerQueue.java

```diff
@@ -68,28 +66,11 @@ static boolean isConsumer(ImageConsumerQueue cqbase, ImageConsumer ic) {
     ImageConsumerQueue(InputStreamImageSource src, ImageConsumer ic) {
         consumer = ic;
         interested = true;
-        // ImageReps do their own security at access time.
-        if (ic instanceof ImageRepresentation) {
-            ImageRepresentation ir = (ImageRepresentation) ic;
-            if (ir.image.source != src) {
-                throw new SecurityException("ImageRep added to wrong image source");
-            }
```

Comment:
This exception also doesn't look directly related to the SM, so we may want to keep it, as you did in `URLImageSource#getDecoder`. 

Removing it changed the behavior, now we can instantiate `ImageConsumerQueue` with a mismatched `ImageRepresentation` image source and `InputStreamImageSource`, but we couldn't before.

Reply:
> This exception also doesn't look directly related to the SM, so we may want to keep it, as you did in `URLImageSource#getDecoder`.
> 
> Removing it changed the behavior, now we can instantiate `ImageConsumerQueue` with a mismatched `ImageRepresentation` image source and `InputStreamImageSource`, but we couldn't before.

Initially I did leave it, then I convinced myself to take it out. So I'm fine with putting it back. And leaving SE with a comment ..